Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,311 @@
+  Scalar theta = electronMass / particle.energy();
+
+  std::uniform_real_distribution<Scalar> uniformDistribution{0., 1.};
+  const Scalar u = -std::log(uniformDistribution(generator) *
+                             uniformDistribution(generator)) *
+                   1.6;
+
+  theta *= (uniformDistribution(generator) < 0.25)
+               ? u
+               : u * 1. / 3.;  // 9./(9.+27) = 0.25
+
+  // draw the random orientation angle
+  const auto psi =
+      std::uniform_real_distribution<double>(-M_PI, M_PI)(generator);
+
+  Acts::Vector3 direction = particle.unitDirection();
+  // construct the combined rotation to the scattered direction
+  Acts::RotationMatrix3 rotation(
+      // rotation of the scattering deflector axis relative to the reference
+      Acts::AngleAxis3(psi, direction) *
+      // rotation by the scattering angle around the deflector axis
+      Acts::AngleAxis3(theta, Acts::makeCurvilinearUnitU(direction)));
+  direction.applyOnTheLeft(rotation);
+  return direction;
+}
+
+std::array<Particle, 2> PhotonConversion::generateChildren(
+    const Particle& photon, Scalar childEnergy,
+    const Particle::Vector3& childDirection) const {
+  // Calculate the child momentum
+  const Scalar massChild = electronMass;
+  const Scalar momentum1 =
+      sqrt(childEnergy * childEnergy - massChild * massChild);
+
+  // Use energy-momentum conservation for the other child
+  const Particle::Vector3 vtmp =
+      photon.fourMomentum().template segment<3>(Acts::eMom0) -
+      momentum1 * childDirection;
+  const Scalar momentum2 = vtmp.norm();
+
+  // Build the particles and store them
+  std::array<Particle, 2> children;
+
+  Particle child1 = Particle(photon.particleId().makeDescendant(0),
+                             static_cast<Acts::PdgParticle>(Acts::eElectron))
+                        .setPosition4(photon.fourPosition())
+                        .setDirection(childDirection)
+                        .setAbsoluteMomentum(momentum1)
+                        .setProcess(ProcessType::ePhotonConversion);
+  children[0] = std::move(child1);
+
+  Particle child2 = Particle(photon.particleId().makeDescendant(1),
+                             static_cast<Acts::PdgParticle>(-Acts::eElectron))
+                        .setPosition4(photon.fourPosition())
+                        .setDirection(childDirection)
+                        .setAbsoluteMomentum(momentum2)
+                        .setProcess(ProcessType::ePhotonConversion);
+  children[1] = std::move(child2);
+
+  return children;
+}
+
+template <typename generator_t>
+bool PhotonConversion::run(generator_t& generator, Particle& particle,
+                           std::vector<Particle>& generated) const {
+  // Fast exit if particle is not a photon
+  if (particle.pdg() != Acts::PdgParticle::eGamma)
+    return false;
+
+  // Fast exit if momentum is too low
+  const Scalar p = particle.absoluteMomentum();
+  if (p < minimumMomentum)
+    return false;
```

Comment:
I think the minimum momentum cut should go, See comment above.

Reply:
As suggested above, the minimum momentum of `2m_e` should be used as threshold. This is now a fixed one.

Reply:
Ok, thanks.